src/base-converter.ts

```diff
@@ -12,109 +12,67 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import * as estree from 'estree';
-import {Iterable as IterableX} from 'ix';
-import * as jsc from 'jscodeshift';
 import {Analysis, Document} from 'polymer-analyzer';
 
+import {ConversionSettings, createDefaultConversionSettings, PartialConversionSettings} from './conversion-settings';
 import {DocumentConverter} from './document-converter';
 import {ConversionResult, JsExport} from './js-module';
 import {ConvertedDocumentFilePath, OriginalDocumentUrl} from './urls/types';
 import {getDocumentUrl} from './urls/util';
-import {getNamespaces} from './util';
 
-export interface BaseConverterOptions {
-  /**
-   * Namespace names used to detect exports. Namespaces declared in the
-   * code with an @namespace declaration are automatically detected.
-   */
-  readonly namespaces?: Iterable<string>;
-
-  /**
-   * Files to exclude from conversion (ie lib/utils/boot.html). Imports
-   * to these files are also excluded.
-   */
-  readonly excludes?: Iterable<string>;
-
-  /**
-   * Namespace references (ie, Polymer.DomModule) to "exclude"" be replacing
-   * the entire reference with `undefined`.
-   *
-   * These references would normally be rewritten to module imports, but in some
-   * cases they are accessed without importing. The presumption is that access
-   * is guarded by a conditional and replcing with `undefined` will safely
-   * fail the guard.
-   */
-  readonly referenceExcludes?: Iterable<string>;
+// NOTE(fks) 11-06-2017: LEGACY!!!
+// Left over from ConversionMetadata, when the settings were mixed in to the
+// converter and its cache/results. Refactored out in the final PR, but left
+// in for now to keep this PR manageable.
+export interface ProjectConverterInterface {
```

Comment:
Once BaseConverter becomes ProjectConverter, this separate interface definition will no longer be needed. We keep it here to satisfy the old `ConversionMetadata` interface.